### PR TITLE
DNN-7719: Improve indexes on Lists table to allow same list name in multiple portals

### PR DESCRIPTION
**Pull request by sleupold**
see Jira ticket for details

### CODE REVIEW

File: Website/Providers/DataProviders/SqlDataProvider/08.01.00.01.SqlDataProvider [hunk 1]
```diff
@@ -166,3 +166,54 @@ AS
...
+GO
+
+-- remove orphaned entries
+DELETE FROM {databaseOwner}[{objectQualifier}Lists] 
+WHERE (DefinitionID NOT IN (SELECT ModuleDefID FROM {databaseOwner}[{objectQualifier}ModuleDefinitions]) AND DefinitionID > 0)
+   OR (PortalID NOT IN (SELECT PortalID FROM {databaseOwner}[{objectQualifier}Portals]) AND PortalID >= 0)
+GO
+
+-- ensure that there are no duplicates before creating the index
+DELETE FROM {databaseOwner}[{objectQualifier}Lists]
```
Comment (IDisposable) [inline Website/Providers/DataProviders/SqlDataProvider/08.01.00.01.SqlDataProvider]: Whoa, that could be a very surprising result of an upgrade attempt... total loss of an earlier (presumably used) list and retention of only the last one added?  I don't know the whole context, obviously... but would it be better to just rename the duplicate lists (ala, the count trick) and then let the end user prune them down after the upgrade?
```
SELECT EntryID, NewName=ListName + COUNT(*)
FROM  {databaseOwner}[{objectQualifier}Lists] 
GROUP BY PortalID, ParentID, ListName, [Value]
```

Comment (sleupold) [inline Website/Providers/DataProviders/SqlDataProvider/08.01.00.01.SqlDataProvider]: no, those items were never been able to be retrieved by the API.

**Comment by galatrash-at-dnn** [comment]
This PR needs to be rebased on the latest source code to be able to process.

**Comment by sleupold** [comment]
I know, it's on my list..

**Comment by sleupold** [comment]
I'll review and resubmit asap

**Comment by mitchelsellers** [comment]
Closing, @sleupold will be submitting a new pull request

**Pull request closed without merging**